### ISSUE TITLE
Type inference error for function overloads

### ISSUE THREAD

**Issue by lkct** [microsoft/pylance-release#925]
## Environment data

-   Language Server version: 2021.2.0
-   OS and version: Windows 10 Pro 1909 18363.720
-   Python version (& distribution if applicable, e.g. Anaconda): 3.8.5 (with PyTorch 1.7.1)

## Expected behaviour

```python
import torch
a = torch.tensor([1])  # a: Tensor
b = torch.tensor([1])  # b: Tensor
c = torch.min(a, b)    # c: Tensor
```

## Actual behaviour

```python
import torch
a = torch.tensor([1])  # inferred a: Tensor
b = torch.tensor([1])  #          b: Tensor
c = torch.min(a, b)    #          c: namedtuple_values_indices
```

## Code Snippet / Additional information

The type inference for `a` and `b` is actually correct, and the only overload of `torch.min()` applicable to two `Tensor` args is the 4th below, but it seems Pylance is choosing a wrong overload.
However, ctrl+click on torch.min actually directs to and highlights the 4th.
```python
# ...\torch\_C\_VariableFunctions.pyi
@overload
def min(input: Tensor, dim: _int, keepdim: _bool=False, *, out: Optional[Tensor]=None) -> namedtuple_values_indices: ...
@overload
def min(input: Tensor, dim: Union[str, ellipsis, None], keepdim: _bool=False, *, out: Optional[Tensor]=None) -> namedtuple_values_indices: ...
@overload
def min(input: Tensor, *, out: Optional[Tensor]=None) -> Tensor: ...
@overload
def min(input: Tensor, other: Tensor, *, out: Optional[Tensor]=None) -> Tensor: ...
```

By explicitly using kwargs, I can force type inference to use the 4th overload, but this should not be expected in my code.
```python
import torch
a = torch.tensor([1])     # inferred a: Tensor
b = torch.tensor([1])     #          b: Tensor
c = torch.min(a, other=b) #          c: Tensor
```

## Logs

```
[Info  - 下午3:44:13] Pylance language server 2021.2.0 (pyright e278f28d) starting
[Info  - 下午3:44:13] Server root directory: c:\Users\lkct\.vscode\extensions\ms-python.vscode-pylance-2021.2.0\dist
[Info  - 下午3:44:13] No configuration file found.
[Info  - 下午3:44:13] Setting pythonPath for service "test": "C:\Users\lkct\AppData\Local\Programs\Python\Python38\python.exe"
Search paths found for configured python interpreter:
  C:\Users\lkct\AppData\Local\Programs\Python\Python38\DLLs
  C:\Users\lkct\AppData\Local\Programs\Python\Python38\lib
  C:\Users\lkct\AppData\Local\Programs\Python\Python38
  C:\Users\lkct\AppData\Roaming\Python\Python38\site-packages
  C:\Users\lkct\AppData\Local\Programs\Python\Python38\lib\site-packages
  C:\Users\lkct\AppData\Local\Programs\Python\Python38\lib\site-packages\win32
  C:\Users\lkct\AppData\Local\Programs\Python\Python38\lib\site-packages\win32\lib
  C:\Users\lkct\AppData\Local\Programs\Python\Python38\lib\site-packages\Pythonwin
[Error - 下午3:44:13] stubPath e:\code\test\typings is not a valid directory.
[Info  - 下午3:44:13] Assuming Python version 3.8
[Info  - 下午3:44:13] Assuming Python platform Windows
[Info  - 下午3:44:13] Searching for source files
[Info  - 下午3:44:13] Found 9 source files
[Info  - 下午3:44:13] Background analysis(1) root directory: c:\Users\lkct\.vscode\extensions\ms-python.vscode-pylance-2021.2.0\dist
[Info  - 下午3:44:13] Background analysis(1) started
Background analysis message: setConfigOptions
Background analysis message: setTrackedFiles
Background analysis message: markAllFilesDirty
Background analysis message: setFileOpened
Background analysis message: getSemanticTokens
[BG(1)] parsing: e:\code\test\test.py (47ms)
[BG(1)] parsing: c:\Users\lkct\.vscode\extensions\ms-python.vscode-pylance-2021.2.0\dist\typeshed-fallback\stdlib\builtins.pyi [fs read 3ms] (122ms)
[BG(1)] binding: c:\Users\lkct\.vscode\extensions\ms-python.vscode-pylance-2021.2.0\dist\typeshed-fallback\stdlib\builtins.pyi (36ms)
[BG(1)] binding: e:\code\test\test.py (1ms)
[BG(1)] parsing: C:\Users\lkct\AppData\Local\Programs\Python\Python38\lib\site-packages\torch\__init__.py [fs read 1ms] (281ms)
[BG(1)] binding: C:\Users\lkct\AppData\Local\Programs\Python\Python38\lib\site-packages\torch\__init__.py ...
[BG(1)]   parsing: C:\Users\lkct\AppData\Local\Programs\Python\Python38\lib\site-packages\torch\_C\__init__.pyi [fs read 2ms] (119ms)
[BG(1)]   binding: C:\Users\lkct\AppData\Local\Programs\Python\Python38\lib\site-packages\torch\_C\__init__.pyi (47ms)
[BG(1)]   parsing: C:\Users\lkct\AppData\Local\Programs\Python\Python38\lib\site-packages\torch\_C\_VariableFunctions.pyi [fs read 0ms] (73ms)
[BG(1)]   binding: C:\Users\lkct\AppData\Local\Programs\Python\Python38\lib\site-packages\torch\_C\_VariableFunctions.pyi (28ms)
[BG(1)]   parsing: C:\Users\lkct\AppData\Local\Programs\Python\Python38\lib\site-packages\torch\functional.py [fs read 0ms] (57ms)
[BG(1)]   binding: C:\Users\lkct\AppData\Local\Programs\Python\Python38\lib\site-packages\torch\functional.py (7ms)
[BG(1)] binding: C:\Users\lkct\AppData\Local\Programs\Python\Python38\lib\site-packages\torch\__init__.py (342ms)
[BG(1)] parsing: c:\Users\lkct\.vscode\extensions\ms-python.vscode-pylance-2021.2.0\dist\typeshed-fallback\stdlib\_typeshed\__init__.pyi [fs read 0ms] (13ms)
[BG(1)] binding: c:\Users\lkct\.vscode\extensions\ms-python.vscode-pylance-2021.2.0\dist\typeshed-fallback\stdlib\_typeshed\__init__.pyi (1ms)
[BG(1)] parsing: c:\Users\lkct\.vscode\extensions\ms-python.vscode-pylance-2021.2.0\dist\typeshed-fallback\stdlib\typing.pyi [fs read 0ms] (29ms)
[BG(1)] binding: c:\Users\lkct\.vscode\extensions\ms-python.vscode-pylance-2021.2.0\dist\typeshed-fallback\stdlib\typing.pyi (8ms)
[BG(1)] parsing: c:\Users\lkct\.vscode\extensions\ms-python.vscode-pylance-2021.2.0\dist\typeshed-fallback\stubs\typing-extensions\typing_extensions.pyi [fs read 0ms] (2ms)
[BG(1)] binding: c:\Users\lkct\.vscode\extensions\ms-python.vscode-pylance-2021.2.0\dist\typeshed-fallback\stubs\typing-extensions\typing_extensions.pyi (1ms)
[BG(1)] parsing: C:\Users\lkct\AppData\Local\Programs\Python\Python38\lib\site-packages\torch\types.py [fs read 1ms] (3ms)
[BG(1)] binding: C:\Users\lkct\AppData\Local\Programs\Python\Python38\lib\site-packages\torch\types.py (1ms)
[BG(1)] parsing: C:\Users\lkct\AppData\Local\Programs\Python\Python38\lib\site-packages\torch\tensor.py [fs read 1ms] (78ms)
[BG(1)] binding: C:\Users\lkct\AppData\Local\Programs\Python\Python38\lib\site-packages\torch\tensor.py (7ms)
Background analysis message: getSemanticTokens
Background analysis message: analyze
[BG(1)] analyzing: e:\code\test\test.py ...
[BG(1)]   checking: e:\code\test\test.py (2ms)
[BG(1)] analyzing: e:\code\test\test.py (2ms)
Background analysis message: resumeAnalysis
IntelliCode model c:\Users\lkct\.vscode\extensions\visualstudioexptteam.vscodeintellicode-1.2.11\cache\E61945A9A512ED5E1A3EE3F1A2365B88F8FE_E4E9EADA96734F01970E616FAB2FAC19
Loading ONNX runtime...
Loaded ONNX runtime. Creating IntelliCode session...
2021-02-07 15:44:15.4228701 [I:onnxruntime:, inference_session.cc:174 onnxruntime::InferenceSession::ConstructorCommon] Creating and using per session threadpools since use_per_session_threads_ is true
2021-02-07 15:44:15.4569683 [I:onnxruntime:, inference_session.cc:840 onnxruntime::InferenceSession::Initialize] Initializing session.
2021-02-07 15:44:15.4571148 [I:onnxruntime:, inference_session.cc:865 onnxruntime::InferenceSession::Initialize] Adding default CPU execution provider.
2021-02-07 15:44:15.4585777 [I:onnxruntime:, reshape_fusion.cc:37 onnxruntime::ReshapeFusion::ApplyImpl] Total fused reshape node count: 0
2021-02-07 15:44:15.4591661 [I:onnxruntime:, reshape_fusion.cc:37 onnxruntime::ReshapeFusion::ApplyImpl] Total fused reshape node count: 0
2021-02-07 15:44:15.4608951 [V:onnxruntime:, inference_session.cc:679 onnxruntime::InferenceSession::TransformGraph] Node placements
2021-02-07 15:44:15.4609772 [V:onnxruntime:, inference_session.cc:681 onnxruntime::InferenceSession::TransformGraph] All nodes have been placed on [CPUExecutionProvider].
2021-02-07 15:44:15.4610729 [V:onnxruntime:, session_state.cc:71 onnxruntime::SessionState::CreateGraphInfo] SaveMLValueNameIndexMapping
2021-02-07 15:44:15.4612011 [V:onnxruntime:, session_state.cc:116 onnxruntime::SessionState::CreateGraphInfo] Done saving OrtValue mappings.
2021-02-07 15:44:15.4686026 [I:onnxruntime:, finalize_session_state.cc:173 onnxruntime::SaveInitializedTensors] Saving initialized tensors.
2021-02-07 15:44:15.4881568 [I:onnxruntime:, finalize_session_state.cc:225 onnxruntime::SaveInitializedTensors] Done saving initialized tensors
2021-02-07 15:44:15.4931577 [I:onnxruntime:, inference_session.cc:954 onnxruntime::InferenceSession::Initialize] Session successfully initialized.
Created IntelliCode session.
```

**Comment by erictraut** [comment]
Thanks for the bug report. The issue here is that the type signature for `min` used an `ellipsis`, which pylance was confusing with a literal `...` and treating as the `Any` type. A fix for this will be in the next release of pylance.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2021.2.1, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202121-10-february-2021
